cli-scripts/get_kura.rb

```diff
@@ -0,0 +1,265 @@
+    meigaras + ["橘屋"]
+  in ["合名会社寒梅酒造", "宮城県"]
+    meigaras + ["鶯咲"]
+  in ["株式会社中勇酒造店", "宮城県"]
+    meigaras + ["花ノ文"]
+  in ["千田酒造株式会社", "宮城県"]
+    meigaras + ["奥鶴"]
+  in ["萩野酒造株式会社", "宮城県"]
+    meigaras + ["日輪田"]
+  in ["株式会社角星", "宮城県"]
+    meigaras + %w[金紋両國 船尾灯]
+  in ["株式会社男山本店", "宮城県"]
+    meigaras + %w[気仙沼男山 美禄]
+  in ["墨廼江酒造株式会社", "宮城県"]
+    meigaras + %w[谷風 弁慶岬]
+  in ["蔵王酒造株式会社", "宮城県"]
+    meigaras + ["ZAO"]
+  in ["森民酒造本家", "宮城県"]
+    meigaras + ["森民"]
+  in ["麓井酒造株式会社", "山形県"]
+    meigaras + ["フモトヰ"]
+  in ["天領酒造株式会社", "岐阜県"]
+    meigaras + ["日野屋"]
+  in ["株式会社平田酒造場", "岐阜県"]
+    meigaras + %w[やまのひかり 山の光]
+  in ["山﨑合資会社", "愛知県"]
+    meigaras + ["尊王"]
+  in ["丸一酒造株式会社", "愛知県"]
+    meigaras + ["ほしいずみ"]
+  in ["水谷酒造株式株式会社", "愛知県"]
+    meigaras + %w[奏 めぐる]
+  in ["渡辺酒造株式会社", "愛知県"]
+    meigaras + ["平勇"]
+  in ["中埜酒造株式会社", "愛知県"]
+    meigaras + ["半田郷"]
+  in ["盛田金しゃちl酒造株式会社", "愛知県"]
+    meigaras + ["初夢桜"]
+  in ["原田酒造合資会社", "愛知県"]
+    meigaras + %w[卯の花 於大の舞 衣が浦若水]
+  else
+    meigaras
+  end
+end
+# rubocop:enable Metrics/MethodLength
+# rubocop:enable Metrics/AbcSize
+
+# 銘柄に空白が含まれうるか
+#
+# @param region [String] 地域
+# @return [Boolean] 銘柄に空白が含まれうるならtrue
+def meigara_include_space?(region)
+  %w[アフリカ オセアニア ヨーロッパ 中南米 北米].include?(region)
+end
+
+# SAKETIMESに載っている銘柄を複数に分解する
+#
+# SAKETIMESに載っている銘柄は以下の状態が含まれる。
+# - 代表銘柄が載っておらず空文字
+# - 代表銘柄が2つ以上あり、空白で区切られている
+# - 代表銘柄自体に空白が含まれる
+# そのため空白で区切っていい場合とだめな場合がある。
+# 現状は海外地域でのみ銘柄に空白が含まれるため、これでHACKしている。
+#
+# @param meigaras [String] SAKETIMESに載っている代表銘柄
+# @param region [String] 地域
+# @return [Array<String>] 複数の代表銘柄に分解した代表銘柄
+def split_meigara(meigaras, region)
+  meigara_include_space?(region) && meigaras != "" ? [meigaras] : meigaras.split
+end
+
+# SAKETIMESの地域ページのテーブルカラムから、蔵データを作成する
+#
+# 帰ってくる代表銘柄は複数がありうるため、配列となっている。
+#
+# @param table_row [Nokogiri::XML::NodeSet] SAKETIMESのテーブルカラムのオブジェクト
+# @param region [String] 地域
+# @return [Hash<Symbol => String, Array<String>>] 蔵名、地域、代表銘柄を持つハッシュ
+def tr_to_data(table_row, region)
+  kura = table_row.css("span.main a")[0].content
+  kura = rename_kura(kura)
+  meigaras = table_row.css("dd")[0].content
+  meigaras = split_meigara(meigaras, region)
+  meigaras = add_meigara(kura, region, meigaras)
+  { kura: kura, region: region, meigaras: meigaras }
+end
+
+# SAKETIMESの1つの地域から、蔵データを取得する
+#
+# @example 実行例
+#   request_kuras([["北海道","https://jp.sake-times.com/sakagura/./hokkaido"]])
+#     #=> [{ kura: "碓氷勝三郎商店", region: "北海道", meigaras: ["北の勝"]},
+#         ...]
+#
+# @param url [String] SAKETIMESの地域の酒蔵ページのURL
+# @param regoin [String] 地域名
+# @return [Array<Hash<Symbol => String>>] 蔵名、地域、代表銘柄持ったハッシュの配列
+def request_kuras(url, region)
+  html = URI.parse(url).open
+  trs = Nokogiri::HTML(html).css("table tr") # テーブルのrowであるtrをすべて取得
+  return [] if trs.empty?                    # 地域に蔵がなければ終了
+
+  trs = trs[1..]                # ヘッダを捨てる
+  trs.reduce([]) { |accum, tr|
+    datas = tr_to_data(tr, region)
+    accum.append(datas)
+  }
```

Comment:
appendするならmapメソッド方が簡単に書けるんじゃない？（試してない）
```suggestion
  trs.map { |tr|
    tr_to_data(tr, region)
  }
```

Reply:
ほんとだ。
ここは古いスクリプトのままだからね。